File: src/Starfield2026.MiniToolbox/src/MiniToolbox.Garc/Models/PocketMonsters/GR.cs
using System.IO;

using MiniToolbox.Garc.Containers;

namespace MiniToolbox.Garc.Models.PocketMonsters
{
    class GR
    {
        /// <summary>
        ///     Loads a GR map model from Pokémon.
        /// </summary>
        /// <param name="data">The data</param>
        /// <returns>The Model group with the map meshes</returns>
        public static RenderBase.OModelGroup load(Stream data)
        {
            RenderBase.OModelGroup models;

            OContainer container = PkmnContainer.load(data);
            models = BCH.load(new MemoryStream(container.content[1].data));

            return models;
        }
    }
}

File: src/Starfield2026.BgEditor/backend/src/lib/ohana/Models/PocketMonsters/GR.ts
/**
 * Loads a GR map model from Pokemon.
 * Ported from OhanaCli.Formats.Models.PocketMonsters.GR (C#).
 */

import { BinaryReader } from '../../Core/BinaryReader.js';
import { PkmnContainer } from '../../Containers/PkmnContainer.js';
import { BCH } from '../BCH/BCH.js';
import { OModelGroup } from '../../Core/RenderBase.js';

export class GR {
  static load(data: BinaryReader): OModelGroup {
    const container = PkmnContainer.load(data);
    return BCH.load(BinaryReader.fromBuffer(container.content[1].data!));
  }
}
